sql/handler.h

```diff
@@ -3835,6 +3838,12 @@ class handler :public Sql_alloc
    notify_table_changed();
  }
 
+/* return false, mean it can't instant alter default */
+virtual bool check_instant_alter(const Alter_inplace_info* inplace_info)
+{
+	return false;
+}
+
```

Comment:
What does the new virtual method actually do, and why can the equivalent logic not be implemented in `ha_innobase::check_if_supported_inplace_alter()` or `ha_innobase::prepare_inplace_alter_table()`?
If you want to change the API, you have to give strong reasons to convince me why it is necessary. I believe that the ALGORITHM=INPLACE API that was introduced in MySQL 5.6 is perfectly suitable for this task.